tests/itest_specs/basic_round_trip/round_trip_for_non_eth_usd.cairo

```diff
@@ -1,2 +1,610 @@
+        let (admin_mydoge_balance_2: Uint256) = IERC20.balanceOf(
+            contract_address=mydoge_addr,
+            account=admin_addr
+        );
+        assert admin_mydoge_balance_2.low = 4906209520949;
+
+        let (admin_mybtc_balance_2: Uint256) = IERC20.balanceOf(
+            contract_address=mybtc_addr,
+            account=admin_addr
+        );
+        assert admin_mybtc_balance_2.low = 49484551;
+
+        let (admin_myusd_balance_2: Uint256) = IERC20.balanceOf(
+            contract_address=myusd_addr,
+            account=admin_addr
+        );
+        assert admin_myusd_balance_2.low = 5000000000;
+
+        // Test inlocked capital in the pools 
+        assert res_long_btc_2.pool_unlocked_capital = 934222658906583790;
+        assert res_long_doge_2.pool_unlocked_capital = 94396381573071755588690;
+
+        // Test balance of option tokens 
+        // FIXME: BTC OPT balance is wrong
+        assert res_long_btc_2.bal_opt = 9999999;
+        assert res_short_btc_2.bal_opt = 9999999;
+        assert res_long_doge_2.bal_opt = 1000000000000;
+        assert res_short_doge_2.bal_opt = 1000000000000;
+
+        // Test pool vol
+        assert res_long_btc_2.pool_volatility = 288230376151711743900;
+        assert res_long_doge_2.pool_volatility = 288230376151711743900;
+        
+        // Test pools position
+        assert res_long_btc_2.opt_long_pos = 0; 
+        assert res_long_btc_2.opt_short_pos = 230584300921369395; 
+        assert res_long_doge_2.opt_long_pos = 0; 
+        assert res_long_doge_2.opt_short_pos = 23058430092136939520000; 
+
+        // Test lpool balance
+        assert res_long_btc_2.lpool_balance = 1164806959827953185;
+        assert res_long_doge_2.lpool_balance = 117454811665208695108690;
+
+        // Test pool locked capital
+        assert res_long_btc_2.pool_locked_capital = 230584300921369395;
+        assert res_long_doge_2.pool_locked_capital = 23058430092136939520000;
+
+        // Test value of pools positions
+        assert res_long_btc_2.pool_position_val = 218696637997535849;
+        assert res_long_doge_2.pool_position_val = 20895712607624488341310;
+
+        ///////////////////////////////////////////////////
+        // WITHDRAW CAPITAL - WITHDRAW 20% of lp tokens
+        ///////////////////////////////////////////////////
+         %{
+            stop_mock_current_price_btc = mock_call(
+                ids.tmp_address, "get_spot_median", [2100000000000, 8, 0, 0]  # mock current BTC price at 21_000
+            )
+        %}
+        let tenth_btc = Uint256(low = 10000000, high = 0);
+        ILiquidityPool.withdraw_liquidity(
+            contract_address=amm_addr,
+            pooled_token_addr=mybtc_addr,
+            quote_token_address=myusd_addr,
+            base_token_address=mybtc_addr,
+            option_type=0,
+            lp_token_amount=tenth_btc
+        );
+        let (res_long_btc_3) = get_stats(input_long_btc);
```

Comment:
Is the plan to assert the Stats? (After the values stop changing when I'm done fiddling with the types)

Reply:
Yes